server2/handlers/user.handler.ts

```diff
@@ -0,0 +1,41 @@
+import {Request, Response} from "express"
+import {IMainController}   from "../controllers/main.controller";
+import {IUser}             from "../../shared/types/Entities/IUser";
+
+
+export const getUsersHandler = async function (this: IMainController, req: Request, res: Response) {
+	try {
+		const users = await this.userController.getUsers()
+
+		return res.json(users).end();
+
+
+	} catch (e) {
+		return res.status(404).send('error' + e)
+	}
+}
+
+export const getUserHandler = async function (this: IMainController, req: Request, res: Response) {
+ try {
+ const user = await this.userController.getOneUser(req.params.id)
+
+ return res.json(user).end()
+ } catch (e) {
+ return res.status(404).send({msg: 'user was not found' + e})
+ }
```

Comment:
this is a good example for a 404

try {
 const user = await this.userController.getOneUser(req.params.id)

 if(!user) {
     return res.status(404).send({msg: 'User does not exist'})
}

return res.json(users).end();

} catch(e) {
    console.error(e)
    return res.status(500).send({msg: 'Something went wrong'})
}